aea/skills/base.py

```diff
@@ -201,11 +193,19 @@ def __init__(self, **kwargs):
         :param skill_context: the skill context
         :param kwargs: keyword arguments
         """
-        self._context = kwargs.pop("skill_context")  # type: SkillContext
-        self._config = kwargs
-        if "name" not in self._config:
+        try:
```

Comment:
this adds more control on the `SkillComponent` input. The demos might break after this change.